src/java.base/share/classes/java/lang/reflect/code/op/ExtendedOps.java

```diff
@@ -2799,15 +2799,35 @@ public List<Body> expressions() {
 
         @Override
         public Block.Builder lower(Block.Builder b, OpTransformer opT) {
+            /*
+            create block for every expression body
+            when yield branch to the next block
+            also create a block for after expressions
+            *
+            * */
+            List<Block.Builder> builders = new ArrayList<>();
+            for (Body expression : expressions()) {
+                builders.add(b.block());
+            }
+            if (expressions().isEmpty()) {
+                builders.add(b);
+            } else {
+                builders.add(b.block());
+            }
+            // if no expression, we should work with b
 
-            Block.Builder[] builders = new Block.Builder[1];
-            builders[0] = b;
+            // for the first expression body, branch to its block
             List<Value> expressions = new ArrayList<>();
             for (Body expression : expressions()) {
-                builders[0].transformBody(expression, List.of(), opT.andThen((block, op) -> {
+                int exprIndex = expressions().indexOf(expression);
+                Block.Builder bb = builders.get(exprIndex);
+                if (exprIndex == 0) {
+                    b.op(branch(bb.successor()));
+                }
```

Comment:
Use a counted loop, rather than using `indexOf` inside the loop. You can hoist you the `b.op(branch(bb.successor()));` and place it in the above `else` branch of code.